saleor/tax/migrations/0004_migrate_tax_classes.py

```diff
@@ -0,0 +1,123 @@
+# Generated by Django 3.2.14 on 2022-07-27 16:05
+
+from collections import defaultdict
+from django.db import migrations
+from django.db.models import Q
+
+
+AVATAX_CODE_META_KEY = "avatax.code"
+AVATAX_DESCRIPTION_META_KEY = "avatax.description"
+VATLAYER_CODE_META_KEY = "vatlayer.code"
+
+TAX_CLASS_ZERO_RATE = "No Taxes"
+
+AVATAX_PLUGIN_ID = "mirumee.taxes.avalara"
+
+BATCH_SIZE = 500
+
+
+def queryset_in_batches(queryset):
+    """Slice a queryset into batches.
+
+    Input queryset should be sorted be pk.
+    """
+    start_pk = 0
+    while True:
+        qs = queryset.filter(pk__gt=start_pk)[:BATCH_SIZE]
+        pks = list(qs.values_list("pk", flat=True))
+        if not pks:
+            break
+        yield pks
+        start_pk = pks[-1]
+
+
+def _populate_tax_class_name_and_metadata(obj):
+    avatax_code = obj.metadata.get(AVATAX_CODE_META_KEY)
+    avatax_description = obj.metadata.get(AVATAX_DESCRIPTION_META_KEY)
+    vatlayer_code = obj.metadata.get(VATLAYER_CODE_META_KEY)
+
+    name = None
+    metadata = {}
+
+    if avatax_code and vatlayer_code:
+        name = avatax_description or avatax_code
+        metadata = {
+            AVATAX_CODE_META_KEY: avatax_code,
+            AVATAX_DESCRIPTION_META_KEY: avatax_description,
+        }
+    else:
+        if avatax_code:
+            name = avatax_description or avatax_code
+            metadata = {
+                AVATAX_CODE_META_KEY: avatax_code,
+                AVATAX_DESCRIPTION_META_KEY: avatax_description,
+            }
+        if vatlayer_code:
+            name = vatlayer_code
+            metadata = {VATLAYER_CODE_META_KEY: vatlayer_code}
+
+    return name, metadata
```

Comment:
If both codes are not None, you're using avatax code. In that case you don't need to have `if else {if & if}` 
```suggestion
    if avatax_code:
        name = avatax_description or avatax_code
        metadata = {
            AVATAX_CODE_META_KEY: avatax_code,
            AVATAX_DESCRIPTION_META_KEY: avatax_description,
        }
    elif vatlayer_code:
        name = vatlayer_code
        metadata = {VATLAYER_CODE_META_KEY: vatlayer_code}
```

Reply:
Yep, thanks Maciej ;)